## Red signal verified — exact #32 bug reproduced on Windows runner

First CI run captured the intended failure. Relevant log excerpt from Windows Python 3.10 (3.11 and 3.12 identical):

\`\`\`
Setting up resumasher at: /d/a/resumasher/resumasher
Creating venv at /d/a/resumasher/resumasher/.venv
Installing dependencies (this can take ~30s on a fast connection, longer on slow wifi)...
install.sh: line 52: /d/a/resumasher/resumasher/.venv/bin/pip: No such file or directory
ERROR: pip install for pip itself failed (exit code 0).
...
##[error]Process completed with exit code 2.
\`\`\`

This is exactly the bug described in #32 (\"install.sh:52 still calls \`\$VENV/bin/pip\` which does not exist on Windows\"). The Windows runner happened to have \`python3\` available via \`actions/setup-python@v6\`, so the python-discovery half of #33's fix isn't exercised in CI — but the \`.venv/bin/\` vs \`.venv/Scripts/\` half is fully reproduced and fails.

### Scorecard

- ✅ Ubuntu 3.10 / 3.11 / 3.12: PASS in 34–41s (unchanged, 286 tests)
- ❌ Windows 3.10 / 3.11 / 3.12: FAIL in 18–23s at `bash install.sh` step
- ✅ Failure is at the CORRECT step (not a YAML / setup misconfig)
- ✅ Failure message matches #32's diagnosis verbatim

### Next

Waiting on #33 to merge. Once it does, rebase this branch → Windows matrix should flip green. If CRLF or PDF round-trip tests surface new issues on Windows, those get separate issues per #34's out-of-scope clause.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -55,3 +55,62 @@ jobs:
           path: /tmp/resumasher-ci-pdfs/
           if-no-files-found: ignore
           retention-days: 7
+
+  test-windows:
+    # The cohort is ~90% Windows, so the Linux-only matrix above covered
+    # ~10% of our actual users. This job runs under Git Bash (MINGW64) — the
+    # same shell cohort students get when they type `bash install.sh` — and
+    # exercises the full install path end-to-end. See issue #34 for context.
+    name: install.sh + pytest (Windows, Python ${{ matrix.python-version }})
+    runs-on: windows-latest
+    defaults:
+      run:
+        # Force Git Bash on every step. windows-latest defaults to PowerShell;
+        # PowerShell does not reproduce the student install experience and
+        # would hide the venv-layout bug this job is meant to catch.
+        shell: bash
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.10", "3.11", "3.12"]
+
+    steps:
+      - name: Use LF line endings on checkout
+        # Windows Git defaults to converting LF to CRLF on checkout, which
+        # makes text-compare tests (and SKILL.md prologue parsing) nondeterministic.
+        # Configure globally BEFORE actions/checkout runs.
+        run: git config --global core.autocrlf input
+
+      - uses: actions/checkout@v6
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v6
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Run install.sh (this is the real test — exercises #32 fix in its natural habitat)
+        # On a fresh Windows runner, `python3` may or may not be present but
+        # `.venv` will always be created under Scripts/, not bin/. Pre-#33
+        # install.sh fails here with "bin/pip: No such file or directory" —
+        # that is the bug, and the red signal is the point.
+        run: bash install.sh
+
+      - name: Verify Windows venv layout
+        # Paranoid assertion: even if install.sh returned 0, confirm the
+        # Scripts/ layout exists. Guards against a future refactor where
+        # install.sh silently degrades to a no-op on Windows.
+        run: |
+          test -x .venv/Scripts/python.exe || { echo "ERROR: .venv/Scripts/python.exe missing after install.sh"; exit 1; }
+          test -x .venv/Scripts/pip.exe || { echo "ERROR: .venv/Scripts/pip.exe missing after install.sh"; exit 1; }
+          echo "Windows venv layout verified: Scripts/python.exe and Scripts/pip.exe both present."
+
+      - name: Install dev dependencies
+        # install.sh only installs runtime deps; dev deps (pytest, etc.) are
+        # installed explicitly so the CI matches the Linux job's coverage.
+        run: .venv/Scripts/pip install -r requirements-dev.txt
+
+      - name: Run pytest
+        # Invoke pytest through the venv Python explicitly (not via PATH) so
+        # we exercise the same path bin/resumasher-exec takes on a real
+        # student's Windows machine.
+        run: .venv/Scripts/python -m pytest -v

--- a/tests/test_prompts.py
+++ b/tests/test_prompts.py
@@ -502,9 +502,23 @@ def skill_tree(tmp_path: Path) -> Path:
 
 
 def _run_build_prompt(*argv: str) -> subprocess.CompletedProcess[str]:
-    """Invoke the orchestration.py CLI as a subprocess."""
+    """Invoke the orchestration.py CLI as a subprocess.
+
+    Pin the decode encoding to UTF-8. The CLI's `if __name__ == "__main__"`
+    block reconfigures stdout/stderr to UTF-8 at write time, so the pipe
+    bytes are always UTF-8. But `subprocess.run(text=True)` on Windows
+    defaults to the system locale (CP1252) when decoding — `ñ` round-trips
+    as `�`, and assertions like `assert "Ana Müller" in r.stdout` fail with
+    mojibake. Force UTF-8 on the read side too.
+    """
     cmd = [sys.executable, "-m", "scripts.orchestration", "build-prompt", *argv]
-    return subprocess.run(cmd, capture_output=True, text=True, check=False)
+    return subprocess.run(
+        cmd,
+        capture_output=True,
+        text=True,
+        check=False,
+        encoding="utf-8",
+    )
 
 
 def test_cli_build_prompt_fit_analyst(skill_tree: Path):

--- a/tests/test_stdout_encoding.py
+++ b/tests/test_stdout_encoding.py
@@ -150,6 +150,18 @@ def test_orchestration_build_prompt_emits_utf8_bytes_for_arrow_glyph(tmp_path):
     )
 
 
+@pytest.mark.skipif(
+    sys.platform == "win32",
+    reason=(
+        "The sanity simulation uses a Linux-style minimal PATH "
+        "(/usr/bin:/bin) and relies on PYTHONIOENCODING overriding a "
+        "default that, on Windows, is already CP1252 — the simulation is "
+        "moot there. Windows already exercises the real CP1252 behavior "
+        "on every stdout write in CI, so regressions of the reconfigure "
+        "fix surface in the two tests above. Skip here to avoid a false "
+        "red from env shape alone."
+    ),
+)
 def test_sanity_cp1252_simulation_reproduces_crash_without_fix():
     """Sanity check: PYTHONIOENCODING=cp1252 actually reproduces the crash.
 

--- a/tests/test_telemetry_scripts.py
+++ b/tests/test_telemetry_scripts.py
@@ -14,9 +14,37 @@ import json
 import os
 import shutil
 import subprocess
+import sys
 from pathlib import Path
 
 import pytest
+
+
+def _resolve_bash() -> str:
+    """Pick the bash interpreter to invoke under subprocess.
+
+    On Windows GitHub Actions runners, `bash` on Windows PATH often resolves
+    to `C:\\Windows\\System32\\bash.exe` (the Windows Subsystem for Linux
+    launcher) BEFORE Git Bash (`C:\\Program Files\\Git\\bin\\bash.exe`). WSL's
+    bash emits "Windows Subsystem for Linux has no installed distributions"
+    to stdout (in UTF-16, no less) and exits non-zero. That's what we hit
+    on the first Windows CI run after adding the bash prefix to the
+    subprocess call — the prefix fixed WinError 193 but let WSL intercept.
+
+    Resolution order:
+      1. `$BASH` env var — set by GitHub Actions when the job runs under
+         `defaults.run.shell: bash` (points at Git Bash on Windows runners).
+      2. Well-known Git Bash path on Windows, if present.
+      3. Plain `bash` — correct on POSIX, and the last resort on Windows.
+    """
+    env_bash = os.environ.get("BASH")
+    if env_bash:
+        return env_bash
+    if sys.platform == "win32":
+        git_bash = Path(r"C:\Program Files\Git\bin\bash.exe")
+        if git_bash.is_file():
+            return str(git_bash)
+    return "bash"
 
 
 REPO_ROOT = Path(__file__).resolve().parents[1]
@@ -28,19 +56,16 @@ SYNC_BIN = REPO_ROOT / "bin" / "resumasher-telemetry-sync"
 def _run(bin_path: Path, args: list[str], env: dict[str, str], cwd: Path | None = None,
          input_text: str | None = None, timeout: int = 15) -> subprocess.CompletedProcess:
     # Invoke through `bash` explicitly rather than letting the OS resolve the
-    # script's shebang. On POSIX this is a no-op — the kernel would honor the
-    # `#!/usr/bin/env bash` shebang anyway. On Windows, Python's CreateProcessW
-    # can only launch PE/EXE files, so `subprocess.run([str(bin_path), ...])`
-    # raises `OSError: [WinError 193] %1 is not a valid Win32 application`.
-    # Git Bash (MINGW64) has its own shebang interpretation layer, but Python
-    # subprocess doesn't use it. Prepending `bash` launches bash.exe (a real
-    # PE binary on Windows, /bin/bash on POSIX) and bash itself handles the
-    # script interpretation. Matches how students actually invoke the scripts
-    # via Git Bash.
+    # script's shebang. On POSIX the kernel would honor `#!/usr/bin/env bash`
+    # directly. On Windows, Python's CreateProcessW can only launch PE/EXE
+    # files — `subprocess.run([str(bin_path), ...])` raises WinError 193.
+    # Git Bash has its own shebang layer but it isn't reached from Python
+    # subprocess. `_resolve_bash()` picks Git Bash explicitly on Windows —
+    # plain `bash` on Windows PATH often resolves to WSL's stub first.
     full_env = os.environ.copy()
     full_env.update(env)
     return subprocess.run(
-        ["bash", str(bin_path), *args],
+        [_resolve_bash(), str(bin_path), *args],
         env=full_env,
         cwd=str(cwd) if cwd else None,
         capture_output=True,

